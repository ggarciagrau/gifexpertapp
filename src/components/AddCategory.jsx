import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        
        let cleanInputValue = inputValue.trim();
        if (!cleanInputValue.length) return;
        onNewCategory(cleanInputValue);
        setInputValue('');
        // setCategories(categories => [inputValue, ...categories]);
    }

    return (
        <div>
            <form onSubmit={onSubmit}>
                <input
                    type="text"
                    placeholder="Search gifs..."
                    value={inputValue}
                    onChange={onChange}
                />
            </form>
        </div>
    );
}