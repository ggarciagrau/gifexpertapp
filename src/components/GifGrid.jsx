import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            { isLoading && <h2>Cargando...</h2> }
            <div className="card-grid">
                {
                    images.map(({ id, title, url }) => (
                        <GifGridItem 
                            key={id}
                            {...{title, url, id}}
                        />
                    ))
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}