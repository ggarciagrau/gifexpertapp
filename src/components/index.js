export * from './AddCategory';
export * from './GifGrid';
export *from './GifGridItem';

// import { AddCategory } from "./AddCategory";
// import { GifGrid } from "./GifGrid";
// import { GifGridItem } from "./GifGridItem";

// export {
//     AddCategory,
//     GifGrid,
//     GifGridItem
// };