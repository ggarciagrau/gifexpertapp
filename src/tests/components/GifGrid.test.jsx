import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../components";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("GifGrid tests", () => {

    const category = "Doge";

    test('Renders (initial loading)', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true
        });

        const { container } = render(<GifGrid category={category} />);
        
        expect(screen.getByText(category));
        expect(screen.getByText('Cargando...'));
        expect(container).toMatchSnapshot();
    });

    test("Must display items when images are loaded with useFetchGifs", () => {

        const gifsFixture = [{ id: "googl", title: "Google", url: "https://google.com"}, { id: "yt", title: "Youtube", url: "https://youtube.com"}];
        useFetchGifs.mockReturnValue({
            images: gifsFixture,
            isLoading: false
        });

        const { container } = render(<GifGrid category={category} />);

        expect(screen.getByText(gifsFixture[0].title));
        expect(screen.getByText(gifsFixture[1].title));
        expect(screen.getAllByRole("img").length).toBe(2);
        expect(container).toMatchSnapshot();
    });
});