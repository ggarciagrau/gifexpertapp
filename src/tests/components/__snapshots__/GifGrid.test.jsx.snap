// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests Must display items when images are loaded with useFetchGifs 1`] = `
<div>
  <h3>
    Doge
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="Google"
        src="https://google.com"
      />
      <p>
        Google
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Youtube"
        src="https://youtube.com"
      />
      <p>
        Youtube
      </p>
    </div>
  </div>
</div>
`;

exports[`GifGrid tests Renders (initial loading) 1`] = `
<div>
  <h3>
    Doge
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
